tests/Core.nut

```diff
@@ -0,0 +1,422 @@
+// MIT License
+//
+// Copyright 2017 Electric Imp
+//
+// SPDX-License-Identifier: MIT
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+// EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+// OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+// ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+// "Promise" symbol is injected dependency from ImpUnit_Promise module,
+// while class being tested can be accessed from global scope as "::Promise".
+
+// Default params for createTest function
+defaultParams = {
+    // Options for Rocky constructor
+    "paramsRocky": {},
+    // Options for calling Rocky.Route constructor directly
+    "paramsRockyRoute": null,
+    // Options for calling Rocky.Context constructor directly
+    "paramsRockyContext": null,
+    // Use for calling Rocky.Context constructor directly. If true, then all Rocky.Context methods will be called.
+    "paramsRockyContextAdditionalUsage": false,
+    // Method to send requests
+    "method": "GET",
+    // Method to receive requests
+    // If not specified, defaultParams.method will be used
+    "methodOverride": null,

```

Comment:
What it this for? Not sure I understand the idea behind this property. Can you elaborate please?